gnovm/pkg/gnolang/stacktrace.go

```diff
@@ -0,0 +1,63 @@
+package gnolang
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Execution struct {
+	Stmt  Stmt
+	Frame *Frame
+}
+type Stacktrace struct {
+	Executions []Execution
+}
+
+func (s Stacktrace) String() string {
+	var builder strings.Builder
+
+	for _, e := range s.Executions {
+		switch {
+		case e.Stmt == nil:
+			builder.WriteString(fmt.Sprintf("%s()\n", e.Frame.Func))
+			builder.WriteString(fmt.Sprintf("    %s/%s:%d\n", e.Frame.Func.PkgPath, e.Frame.Func.FileName, e.Frame.Func.Source.GetLine()))
+		case e.Frame.Func != nil && e.Frame.Func.IsNative():
+			builder.WriteString(fmt.Sprintf("%s()\n", e.Frame.Func))
+			builder.WriteString(fmt.Sprintf("    %s.%s\n", e.Frame.Func.NativePkg, e.Frame.Func.NativeName))
+		case e.Frame.Func != nil:
+			builder.WriteString(fmt.Sprintf("%s\n", e.Stmt.String()))
+			builder.WriteString(fmt.Sprintf("    %s/%s:%d\n", e.Frame.Func.PkgPath, e.Frame.Func.FileName, e.Stmt.GetLine()))
+		default:
+			builder.WriteString(fmt.Sprintf("%s\n", e.Frame.GoFunc.Value.Type()))
+			builder.WriteString("    gonative\n")
+		}
+	}
+	return builder.String()
+}
+
+func (m *Machine) Stacktrace() Stacktrace {
+	stacktrace := Stacktrace{}
```

Comment:
I have a nit pick -- this can just take the default value; no assignment necessary.
```suggestion
	var stacktrace Stacktrace
```

Reply:
what do you think to use named returns ?

Reply:
I don't mind them, but I know many developers have a strong opinion about them 😄